File: lab07/log_probability.pl
#!/usr/bin/perl -w
if ($#ARGV != 0){
    print("Please give one word!\n");
    exit(0);
}
$word = lc($ARGV[0]);
foreach $file (glob "poems/*.txt") {
    open(F, $file);
    @f = <F>;
    $wcount = 0;
    $tcount = 0;
    foreach $l (@f){
        $l =~ s/'s/ s/g;
        $l =~ s/[^a-zA-Z]/ /g;
        @line = split/\s+/, lc($l);
	foreach $w (@line){
	    if ($w ne "") {
		$tcount++;
	    }
	    if ($w =~ /^$word$/){
		$wcount++;
	    }
	}
    }
    $file =~ s/poems\/|.txt//g;
    $file =~ s/_/ /g;
    printf("log((%d+1)/%6d) = %8.4f %s\n",$wcount,$tcount,log($wcount+1)-log($tcount),$file);

}

File: lab07/log_probability.py
#!/usr/bin/python3
#!/usr/bin/python3
import sys
import glob
import re
from math import *
r = []
for file in glob.glob("poems/*.txt"):
    r.append(file)
r.sort()
for file in r:
    F = open(file, 'r')
    w_count = 0
    t_count = 0
    line = F.readline()
    while line:
        line = line.lower()
        line = re.sub(r'[^a-z]',' ',line)
        line = re.sub(r'\'s',' s',line)
        words = re.split(r'\s+',line)
        for w in words:
            if w != '':
                t_count +=1
            if w == sys.argv[1].lower():
                w_count += 1
        line = F.readline()
    file = re.sub(r'_',' ',file)
    file = re.sub(r'.txt|poems\/','',file)
    print("log(({:d}+1)/{:6d}) = {:8.4f} {}".format(w_count,t_count,log(w_count+1) - log(t_count),file))
